src/internal.rs

```diff
@@ -295,8 +295,8 @@ pub mod auth_v2 {
         pub(in crate::internal::auth_v2) timestamp: OffsetDateTime,
     }
 
-    impl<'a> AuthV2Builder<'a> {
-        pub fn new(req_auth: &'a RequestAuth, timestamp: OffsetDateTime) -> AuthV2Builder {
+    impl AuthV2Builder<'_> {
```

Comment:
This seems suspicious.

Reply:
Just a clippy lint about eliding the lifetime